coverage_point_calculator/src/location.rs

```diff
@@ -0,0 +1,179 @@
+use rust_decimal::Decimal;
+use rust_decimal_macros::dec;
+
+use crate::RadioType;
+
+const RESTRICTIVE_MAX_DISTANCE: Meters = Meters(50);
+
+#[derive(Debug, Clone, PartialEq, PartialOrd)]
+pub struct Meters(u32);
+
+impl Meters {
+    pub fn new(meters: u32) -> Self {
+        Self(meters)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct LocationTrustScores {
+    pub multiplier: Decimal,
+    pub trust_scores: Vec<LocationTrust>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct LocationTrust {
+    pub distance_to_asserted: Meters,
+    pub trust_score: Decimal,
+}
+
+impl LocationTrustScores {
+    pub fn new(radio_type: &RadioType, trust_scores: Vec<LocationTrust>) -> Self {
+        // CBRS radios are always trusted because they have internal GPS
+        let multiplier = if radio_type.is_cbrs() {
+            dec!(1)
+        } else {
+            multiplier(&trust_scores)
+        };
+
+        Self {
+            multiplier,
+            trust_scores,
+        }
+    }
+
+    pub fn new_with_boosted_hexes(
+        radio_type: &RadioType,
+        trust_scores: Vec<LocationTrust>,
+    ) -> Self {
+        let trust_scores: Vec<_> = trust_scores
+            .into_iter()
+            .map(LocationTrust::into_boosted)
+            .collect();
+
+        Self::new(radio_type, trust_scores)
+    }
+}
+impl LocationTrust {
+    fn into_boosted(self) -> Self {
+        // Cap multipliers to 0.25x when a radio covers _any_ boosted hex
+        // and it's distance to asserted is above the threshold.
+        let trust_score = if self.distance_to_asserted > RESTRICTIVE_MAX_DISTANCE {
+            dec!(0.25).min(self.trust_score)
+        } else {
+            self.trust_score
+        };
+
+        LocationTrust {
+            trust_score,
+            distance_to_asserted: self.distance_to_asserted,
+        }
+    }
+}
+
+fn multiplier(trust_scores: &[LocationTrust]) -> Decimal {
+    let count = Decimal::from(trust_scores.len());
+    let scores: Decimal = trust_scores.iter().map(|l| l.trust_score).sum();
+
+    scores / count
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn boosted_hexes_within_distance_retain_trust_score() {
```

Comment:
To fully test the boundary, shouldn't we have also have a test for 50m??

Reply:
The Location tests were rewritten to be more scenario forward. 
And the values not include the boundary limits directly.